File: java/program22.java
import java.util.Scanner;

class ArrayX
{
    protected int Arr[];

    public ArrayX(int iSize)
    {
        Arr = new int[iSize];
    }

    protected void Accept()
    {
        int iCnt = 0;

        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter elements of array :");

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }
    }

    protected void Display()
    {
        System.out.println("Element of the array are :");

        for(int iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            System.out.println(Arr[iCnt]+"\t");
        }
        System.out.println();
    }
}

class MyArray extends ArrayX
{
    public MyArray(int iSize)
    {
        super(iSize);
    }

    public int CheckFirstOcc(int iSearch)
    {
        int iCnt = 0;
        int iPos = -1;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] == iSearch)
            {
                iPos = iCnt;
                break;
            }
        }
        return iPos;
    }
}

class program22
{
    public static void main(String arg[])
    {
        int iValue = 0,iValue1 = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of elements ");
        iValue = sobj.nextInt();

        MyArray obj = new MyArray(iValue);
        obj.Accept();
        obj.Display();
        
        System.out.println("Enter number to search :");
        iValue1 = sobj.nextInt();
        int iRet = obj.CheckFirstOcc(iValue1);
        if(iRet == -1)
        {
            System.out.println("There is no such number in array :");
        }
        else
        {
            System.out.println("First occurence of the element is at index : "+iRet);
        }
    }
}

File: c#/program22.cs
using System;

class ArrayX
{
    public int[] Arr;

    public ArrayX(int size)
    {
        this.Arr = new int[size];
    }

    internal void Accept()
    {
        Console.WriteLine("Enter the elements of the array :");

        for(int iCnt = 0; iCnt < Arr.Length; iCnt++)
        {
            Arr[iCnt] = int.Parse(Console.ReadLine());
        }
    }

    internal void Display()
    {
        Console.WriteLine("Elements of the array are :");

        for(int iCnt = 0; iCnt < Arr.Length; iCnt++)
        {
            Console.Write(Arr[iCnt]+"\t");
        }
        Console.WriteLine();
    }
}

class MyArray : ArrayX
{
    public int ERR_NOTUFOUND;

    public MyArray(int size) : base(size)
    {
        this.ERR_NOTUFOUND = -1;    
    }

    public int CheckFirstOcc(int search)
    {
        int iCnt = 0;

        for(iCnt = 0; iCnt < Arr.Length; iCnt++)
        {
            if(Arr[iCnt] == search)
            {
                break;
            }
        }
        if(iCnt == this.Arr.Length)
        {
            return ERR_NOTUFOUND;
        }        
        else 
        {
            return iCnt;
        }
    }
}

class program22
{
    private static void Main(string[] args)
    {
        int size = 0, ret = 0, value = 0; 

        Console.WriteLine("Enter the size of the array :");
        size = int.Parse(Console.ReadLine());

        MyArray obj = new MyArray(size);
        obj.Accept();
        obj.Display();

        
        Console.WriteLine("Enter the element to search");
        value = int.Parse(Console.ReadLine());

        ret = obj.CheckFirstOcc(value);
        if(ret != -1)
        {
            Console.WriteLine("Element found at index = {0}",ret);
        }
        else
        {
            Console.WriteLine("There is no such element");
        }
    }
}